File: hc-sd-web/src/main/webapp/resources/app/view/sdstylebox/SdStyleBoxController.js
/**
 * Description: 款号与外箱关系表Controller
 * All rights Reserved, Designed ByHc
 * Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     yinlp
 * @date:  2015-03-25 08:45:46
 * @version 1.0.0
 */
Ext.define('Hc_sd.view.sdstylebox.SdStyleBoxController', {
    extend: 'Hc_Common.view.BaseSimplePageController',
    alias: 'controller.sdstylebox',
    init:function(){
    	var me = this,objList= me.getObjList();
    	me.callParent(arguments);
    	me.workObject.on('edit','onGridAfterEdit',me);
    },
    
    onGridAfterEdit:function(editor, obj)
    {
    	if(obj.field == 'materialno') {
			var record = obj.record;
			var materialNo = record.get('materialno');
			var sstore = Ext.create('Hc_Common.store.Base', {
			    fields: ["materialno", "materialname"],
                proxy:{
                    url:Hc.mdmPath+"bas_material/list.json"
                }
			});
			
			sstore.proxy.extraParams["materialno"]=materialNo;
			sstore.reload();
			sstore.reload({
			    scope: this,
			    callback: function(records, operation, success) {
			    	if (records.length>0)
			    	{
			    		obj.record.set('materialname',records[0].data.zoneName);
			    	}
			    	else
			    	{
			    		obj.record.set('materialname',"");
			    	}
			    }
			});
		}
    }
  });

File: hc-sd-web/src/main/java/com/hc/scm/sd/web/controller/SdStyleBoxController.java
package com.hc.scm.sd.web.controller;

import com.hc.scm.common.base.service.BaseCrudService;
import com.hc.scm.common.base.web.BaseCrudController;
import com.hc.scm.sd.dao.entity.SdStyleBox;
import com.hc.scm.sd.service.SdStyleBoxService;
import javax.annotation.Resource;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

/**
 * Description: 请写出类的用途
 * All rights Reserved, Designed By hc* Copyright:   Copyright(C) 2014-2015
 * Company:     Wonhigh.
 * @author:     Administrator
 * @date:  2015-04-20 10:32:37
 * @version 1.0.0
 */
@Controller
@RequestMapping("/sd_style_box")
public class SdStyleBoxController extends BaseCrudController<SdStyleBox> {
    @Resource
    private SdStyleBoxService sdStyleBoxService;

    @Override
    public BaseCrudService init() {
        return sdStyleBoxService;
    }
}